src/main/kotlin/inc/evil/d17map/controllers/ReservationController.kt

```diff
@@ -279,4 +280,92 @@ class ReservationController(private val reservationService: ReservationService)
         val reservations = reservationService.getAllReservationsForBuilding(buildingName)
         return ResponseEntity(reservations, HttpStatus.OK)
     }
+
+    @GetMapping("$BUILDING_PATH/recurringReservations/{recurringId}")
+    @Operation(
+        summary = "Get all reservations in a recurring cycle",
+        description = "Retrieve all reservations (upcoming and past, excluding deleted) for a specific recurring cycle.",
+        responses = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved reservations"),
+            ApiResponse(responseCode = "404", description = "Recurring reservation not found")
+        ]
+    )
+    fun getAllReservationsInCycle(
+        @PathVariable buildingName: String,
+        @PathVariable recurringId: UUID
+    ): List<ReservationResponse> {
+        return reservationService.getAllReservationsInCycle(buildingName, recurringId)
+    }
+
+    @DeleteMapping("$BUILDING_PATH/recurringReservations/{recurringId}")
+    @Operation(
+        summary = "Remove upcoming reservations in a recurring cycle",
+        description = "Delete all upcoming reservations for a specific recurring cycle. Past reservations remain unaffected.",
+        responses = [
+            ApiResponse(responseCode = "204", description = "Successfully deleted reservations"),
+            ApiResponse(responseCode = "404", description = "Recurring reservation not found")
+        ]
+    )
+    fun removeUpcomingReservationsInCycle(
+        @PathVariable buildingName: String,
+        @PathVariable recurringId: UUID
+    ) {
+        reservationService.removeUpcomingReservationsInCycle(buildingName, recurringId, removeOnlyUpcoming = true)
+    }
+
+    @PostMapping("$BUILDING_PATH/recurringReservations")
+    @Operation(
+        summary = "Add reservations in a recurring cycle",
+        description = "Create all reservations for a specific recurring cycle.",
+        responses = [
+            ApiResponse(responseCode = "204", description = "Successfully added recurring reservations"),
+            ApiResponse(responseCode = "404", description = "Recurring reservation not added.")
+        ]
+    )
+    fun createRecurringReservation(
+        @PathVariable buildingName: String,
+        @RequestBody reservationRequest: ReservationRequest
+    ): ResponseEntity<Map<String, Any>> {
+        val result = reservationService.createRecurringReservation(buildingName, reservationRequest)
+        return ResponseEntity.ok(result)
+    }
+
+
+    @PostMapping("$BUILDING_PATH/recurringReservations/skip-collisions")
```

Comment:
Co wy na to, żeby zamiast robić osobny endpoint dać `skip-collistions` jako `@RequestParam`, czyli po prostu mieć parametr w URL o nazwie `skip-collistions`. Jeśli dobrze rozumiem, to można zrobić to tak, że sama obecność parametru o takiej nazwie już nam wystarczy, ale można podejść do tego standardowo, bezpiecznie i po prostu przekazywać w formie boolean:

`/buildings/{id}/recurringReservations?skip-collisions=true`

Nasza logika backendowa byłaby wtedy taka, że jeśli ten parametr nie jest nullem (i/lub jeśli jest na true), to wtedy wykonujemy logikę skipowania, a w przeciwnym wypadku podejmujemy próbę rezerwacji.